addon/services/file-manager.js

```diff
@@ -141,25 +244,65 @@ export default Ember.Service.extend({
                 headers: {
                     Authorization: `Bearer ${accessToken}`
                 },
-                success: (data) => resolve(data),
-                error: (_, __, error) => reject(error)
+                error: (_, __, error) => reject(error),
+                success: (data) => resolve(data)
             });
         });
     },
 
-    _pushToStore(data, id, parentFolder) {
-        // Hack the Waterbutler entity to look like the file Ember expects
-        data.data.id = id;
-        data.data.type = 'file';
-        let attr = data.data.attributes;
-        attr.materializedPath = attr.materialized;
-        attr.dateModified = attr.modified;
+    /**
+     * Get an updated `file` model from the OSF API.
+     *
+     * @method _reloadModel
+     * @private
+     * @param {Object} model Ember model to reload.
+     * @param {Function} isFresh Function that is called with the reloaded
+     * model from the OSF API. If the model is actually still stale, should
+     * return `false`. Otherwise, return the fresh model.
+     * @return {Promise} Promise that resolves to an updated `file` model or
+     * rejects with an error message.
+     */
+    _reloadModel(model, isFresh) {
+        const maxTries = 32;
```

Comment:
Can you clarify the purpose of the interval? 

In particular: 0.256 seconds isn't a long time to wait between attempts, and 32 retries is an extremely large number if we're talking about server traffic. 

If you're waiting for an event to occur, never use time as a proxy; tie the behavior to resolution of a promise instead. But I'm guessing the issue is getting waterbutler to register data has changed on its end- if that's the case it may be time to ping Fitz again for a better workaround.